tests/e2e/lib/diskutils

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+#
+# Copyright 2023 The qm Authors
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; If not, see <http://www.gnu.org/licenses/>.
+
+select_disk_to_partition(){
+  ###########################################################################
+  # Description:                                                            #
+  # Select free partition with size in "G"                                  #
+  #                                                                         #
+  # Arguments: should be be exported by calling script                      #
+  #   DEVICE,  set free /dev/ device                                        #
+  #   PART_ID, set device partition                                         #
+  ###########################################################################
+
+  local disk_table
+  disk_table=$(lsblk --noheadings --raw)
+  local disks_arr
+  disks_arr=$(echo "$disk_table" | awk '$1~// && $6=="disk" {print $1}')
+  info_message "select_disk_to_partition, found ${disks_arr}"
+  info_message "=============================="
+
+  for disk in $disks_arr; do
+     if [[ ${disk} == "vda" || \
+           $(echo "${disk_table}" | grep -c "${disk}" ) -eq 1 && ${disk} != "zram0" ]];then
+         new_part="$( (echo n; echo p; echo ; echo ; echo ; echo w) | fdisk "/dev/${disk}")"
+         part_id="$(echo "${new_part}" | grep -Po "new partition \K([0-9])")"
+         DISK="${disk}"
+         if [[ $(echo "${disk}" | grep -c nvme) -eq 1 ]]; then
+           part_id=p${part_id}
+           PART_ID="${part_id}"
+           info_message "select_disk_to_partition, found /dev/${DISK}${PART_ID}"
+           info_message "=============================="
+           break;
+         fi
+      fi
+      if [ -e /sys/devices/soc0/machine ]; then
```

Comment:
will be always soc0 and sde?

Reply:
I believe it's possible to detect such machine via cat /etc/machine-id OR sudo dmidecode -s system-uuid

Reply:
not in the RideSXchip
 dmidecode -s system-uuid return empty result
 [root@localhost ~]# dmidecode 
# dmidecode 3.6
# No SMBIOS nor DMI entry point found, sorry.

Booted by android,
I will ignore this one

Reply:
last question before the merge @Yarboa, whats the output of:
```
sudo grep -i "ridesx4" /sys/class/dmi/id/*

```

Reply:
```
 grep -ri "ridesx4" /sys/class/dmi/id/*
grep: /sys/class/dmi/id/*: No such file or directory

```

but you have this 
```
cat /etc/machine-id 
506b9b0ac0c6476ab57bc5c553cf4a4a

```


I suspect it is related to aboot

Reply:
/etc/machine-id should be unique per machine, do not help us in this case.